File: Go/BinarySearch/binarySearch.go
package main

import "fmt"

func binarySearch(arr []int, ele int) int {
	var l int = 0
	var r int = len(arr) - 1

	for l <= r {
		var m int = l + (r-l)/2

		if (arr[m] == ele) {
			return m
		} else if (arr[m] < ele ) {
			l = m + 1
		} else {
			r = m - 1
		}
	}

	return -1
}

func main() {
	var n int;
	fmt.Print("Enter number of elements: ")
	fmt.Scan(&n)

	arr := make([]int, n)

	for i := 0; i<n; i++ {
		fmt.Print("Enter element ",i,": ")
		fmt.Scan(&arr[i])
	}

	fmt.Print("Array elements: ")
	for _, val := range arr {
		fmt.Print(val, " ")
	}
	fmt.Println()

	var ele int;
	fmt.Print("Enter element to search: ")
	fmt.Scan(&ele)

	pos := binarySearch(arr, ele)

	if pos < 0 {
		fmt.Println("Could not find element in array")
	} else {
		fmt.Printf("Element found at position %d\n", pos)
	}
}

File: C++/binarySearch.cpp
#include<iostream>
#include<vector>
using namespace std;

int binarySearch(vector<int> &arr, int element) {
    int l = 0;
    int r = arr.size() - 1;


    while(l <= r){
        int m = (l + (r-l)/2);

        if(arr[m] == element) return m;

        if (arr[m] < element) l = m + 1;

        else r = m - 1;
    }
    return -1;
}

int main() {

    vector<int> arr;
    int n;

    cout<<"Enter number of elements: ";
    cin>>n;

    cout<<"Enter elements in sorted order"<<endl;
    int data;
    for(int i=0; i<n; i++){
        cout<<"Element "<<i<<": ";
        cin>>data;
        arr.push_back(data);
    }
    
    cout<<"Array elements: ";
    for(int i=0; i<n; i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;

    int ele, pos;
    cout<<"Enter element to search: ";
    cin>>ele;

    pos = binarySearch(arr, ele);

    if(pos >= 0) cout<<"Found element at position "<<pos;

    else cout<<"Could not find element in array";

    return 0;
}